<div style="border:solid green 2px; padding: 20px">
<b>Привет, Дмитрий!</b>

Меня зовут Алан, и я буду проверять твой проект. Предлагаю общаться на «ты» :) Но если это не удобно - дай знать, и мы перейдем на "вы". 

Моя основная цель — не указать на совершенные тобою ошибки, а поделиться своим опытом и помочь тебе стать аналитиком данных. Тобой уже проделана большая работа над проектом, но давай сделаем его еще лучше. Ниже ты найдешь мои комментарии - пожалуйста, не перемещай, не изменяй и не удаляй их. Я буду использовать цветовую разметку:

<div class="alert alert-danger">
<b>Комментарий ревьюера:</b> Так выделены самые важные замечания. Без их отработки проект не будет прният. </div>

<div class="alert alert-warning">
<b>Комментарий ревьюера:</b> Так выделены небольшие замечания, либо рекоммендации к улучшению. Одна-два таких замечания в проекте допустимы, но если их будет больше - тебе будет необходимо внести исправления. Это как тестовое задание при приеме на работу: очень много мелких ошибок могут стать причиной отказа кандидату. 

</div>

<div class="alert alert-success">
<b>Комментарий ревьюера:</b> Так я выделяю все остальные комментарии.</div>

Давай работать над проектом в диалоге: если ты что-то меняешь в проекте или отвечаешь на мои комменатри — пиши об этом. Мне будет легче отследить изменения, если ты выделишь свои комментарии:
<div class="alert alert-info"> <b>Комментарий студента:</b> Например, вот так.</div>

Всё это поможет выполнить повторную проверку твоего проекта оперативнее. 

## Исследование надёжности заёмщиков

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.

Результаты исследования будут учтены при построении модели **кредитного скоринга** — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

<div class="alert alert-danger">

**Комментарий ревьюера:** 
Немного про оформление:) Ты просто скопировал задание себе в тетрадку. В некоторых случаях это тоже выход:) Но я бы порекомендовал тебе составлять план, похожий больше на книжное содержание.
Например:

### Определение чегототам
_____
##### В данном исследовании представлены несколько таблиц с блаблабла
______

###### Данное исследование разделим на несколько частей.

##### Часть 1. Изучение общей информации:
* [1. Изученеие файлов с данными, получение общей информации, загрузка библиотек.](#1-bullet)
* [2. Нахождение и ликвидация пропусков.](#2-bullet)

##### Часть 2. Подготовка данных:
* [1. Приведение данных к нужным типам.](#3-bullet)
* [2. Нахождение и исправление аномалий и ошибок.](#4-bullet)

и т.д

Старайся выполнять свой проект в порядке, прописанном в твоем плане. Да, такое оформление займет больше времени, но позволит тебе не путаться в будущем:)
_________________

Обрати внимание, что пункты в частях у меня в виде ссылок. Это очень удобно, когда нужно переходить по разным частям проекта.
Вот тут почитай, как сделать <a href="https://sebastianraschka.com/Articles/2014_ipython_internal_links.html">кликабельные разделы</a>

**Почему это важно?** Когда ты будешь в будущем выполнять тестовые задания, просто скопировав присланную тебе задачу и не составив плана, ты можешь смутить будущего работодателя. А грамотно сконфигурированный план покажет тебя как ответсвенного и усидчивого специалиста. И, просто поверь мне, с расписанием легче:)

<div class="alert alert-info"> 

*Привет, Алан!*

Честно говоря даже не копировал тетрадку, а просто открыл проект по ссылке в тренажере.
К сожалению нигде не увидел, что нам необходимо изучить Markdown. Посмотрел информацию в интеренете, плюс как пример возьму твои заметки. Надеюсь этого будет достаточно.

</div>

<div class="alert alert-success">

**Комментарий ревьюера v2:** Да так гораздо лучше 👍 
</div>

<div class="alert alert-info">
    
### Исследование надежности заемщиков
_____
##### В данном исследовании представлена таблица платежеспособности клиентов банка

______

###### Данное исследование разделим на несколько частей.

##### Часть 1. Изучение общей информации:
* [1. Изученеие файлов с данными, получение общей информации, загрузка библиотек.](#1-bullet)


##### Часть 2. Подготовка данных:
* [1. Нахождение и ликвидация пропусков.](#2-bullet)
* [2. Приведение данных к нужным типам.](#3-bullet)
* [3. Нахождение и исправление аномалий и ошибок.](#4-bullet)
* [4. Стемминг и лемматизация.](#5-bullet)
* [5. Категоризация данных.](#6-bullet)

##### Часть 3. Ответы на вопросы:
* [1. Зависимость между наличием детей и возвратом кредита в срок.](#7-bullet)
* [2. Зависимость между семейным положением и возвратом кредита в срок.](#8-bullet)
* [3. Зависимость между уровнем дохода и возвратом кредита в срок.](#9-bullet)
* [4. Зависимость между целью кредита и возвратом кредита в срок.](#10-bullet)

##### Часть 4. Общий вывод:
* [Ответ на целевой вопрос](#11-bullet)
</div>

<a id='1-bullet'></a>

### Шаг 1. Откройте файл с данными и изучите общую информацию. 

In [23]:
import pandas as pd
from nltk.stem import SnowballStemmer 
russian_stemmer = SnowballStemmer('russian')

df = pd.read_csv('/datasets/data.csv')

In [24]:

print(df.info())
display(df.head(10))

#Проверим уникальные значения в столбцах с категорийными переменными
print(df['children'].value_counts())
print(df['education'].value_counts())
print(df['family_status'].value_counts())
print(df['gender'].value_counts())
print(df['income_type'].value_counts())
print(df['debt'].value_counts())
print(df['purpose'].value_counts())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
children            21525 non-null int64
days_employed       19351 non-null float64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        19351 non-null float64
purpose             21525 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB
None


children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   Среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   
5         0    -926.185831         27    высшее             0   
6         0   -2879.202052         43    высшее             0   
7         0    -152.779569         50   СРЕДНЕЕ             1   
8         2   -6929.865299         35    ВЫСШЕЕ             0   
9         0   -2188.756445         41   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   
5  гражданский брак                 1      M   компаньон     0  255763.565419   
6   женат / замужем                 0      F   компаньон     0  240525.971920   
7   женат / замужем                 0      M   сотрудник     0  135823.934197   
8  гражданский брак                 1      F   сотрудник     0   95856.832424   
9   женат / замужем                 0      M   сотрудник     0  144425.938277   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу  
5               покупка жилья  
6           операции с жильем  
7                 образование  
8       на проведение свадьбы  
9     покупка жилья для семьи

 0     14149
 1      4818
 2      2055
 3       330
 20       76
-1        47
 4        41
 5         9
Name: children, dtype: int64
среднее                13750
высшее                  4718
СРЕДНЕЕ                  772
Среднее                  711
неоконченное высшее      668
ВЫСШЕЕ                   274
Высшее                   268
начальное                250
Неоконченное высшее       47
НЕОКОНЧЕННОЕ ВЫСШЕЕ       29
НАЧАЛЬНОЕ                 17
Начальное                 15
ученая степень             4
УЧЕНАЯ СТЕПЕНЬ             1
Ученая степень             1
Name: education, dtype: int64
женат / замужем          12380
гражданский брак          4177
Не женат / не замужем     2813
в разводе                 1195
вдовец / вдова             960
Name: family_status, dtype: int64
F      14236
M       7288
XNA        1
Name: gender, dtype: int64
сотрудник          11119
компаньон           5085
пенсионер           3856
госслужащий         1459
предприниматель        2
безработный           

### Вывод

Данные представляют собой таблицу из 21525 строк и 12 столбцов
1. Типы данных в столбцах на первый взляд соответвуют содержимому, кроме столбца debt. Его можно заменить на boolean
2. Отрицательные значения в столбце days_employed. Разный регистр в столбце education
3. Пустые значения в столбцах days_employed и total_income. На первый взгляд логичное совпадение количества пустых значений в этих столбцах. Гипотеза: те у кого нет работы, не имеют общего стажа и дохода, т.е. NaN будет в одинаковых строках по этим двум столбцам.
4. Гипотезу из п.3 можно проверить по статусам слобца income_type. Методом value_counts() посчитаны количество различных статусов столбца income_type. К сожалению гипотеза п.3 тут не подтверждается. Необходим детальный анализ (произведем ниже).
5. Странные данные в столбце children. 76 человек имеют 20 детей и 47 человек по -1 ребенку. Данных немного, поэтому можно попробовать исключить их из расчета либо заменить на средние значения.
6. В столбце purpose похожие цели описаны разными фразами. Можно применить лемматизацию для унификации списка.

<div class="alert alert-warning">

**Комментарий ревьюера:** 
* Библиотеки, лучше стараться импортировать в отдельной ячейке для ясности и удобства. Например, если они у тебя были в одной ячейке с загрузкой файла, то тебе при добавлении новой библиотеки в ходе работы больше не придется по новой загружать файл.
* Иногда лучше использовать `display()` вместо `print()`, чтобы таблицы выводились в читабельном виде.
</div>

### Шаг 2. Предобработка данных

<a id='2-bullet'></a>

### Обработка пропусков

In [35]:
#Проверяем гипотезу п.3. Создадим таблицу, где значения days_employed будут пустые
df_de_nan = df[df['days_employed'].isnull()]
display(df_de_nan)
#На первый взгляд пропуски в обоих столбцах совпадают. Проверим количество строк NaN в столбце total_income при выставленном фильтре NaN в столбце days_employed
print(df_de_nan.count())
#Ни одного заполненного значения, значит строки пустые строки в стоблцах 'days_employed','total_income' совпадают
#Проверим тип занятости людей с пустыми значениями стажа и зарплаты
print(df_de_nan['income_type'].value_counts())
#Корреляции в отсутствии дохода с типом занятости нет. Проверка по другим категориям также не дает понимания
print(df_de_nan['children'].value_counts())

#Т.к. нет явных причин отсутствия данных в столбцах доход и стаж можно попробовать заполнить пропуски по виду занятости
#Возьмем медианное значение по столбцу "total_income" для каждой категории столбца "income_type"
#Для "days_employed" среднее для каждой категории столбца "dob_years"
#Чтобы не переписывать датафрейм создадим новый, т.к. далее не планируется использовать данные из ячеек с пустыми "total_income"
# и "days_employed"
df_no_emplties = df
df_no_emplties['total_income'] = df['total_income'].fillna(df.groupby('income_type')['total_income'].transform('median'))
#Предварительно заменим отрицательные значения в столбце "days_employed"
df_no_emplties['days_employed'] = df['days_employed'].abs()
df_no_emplties['days_employed'] = df['days_employed'].fillna(df.groupby('dob_years')['days_employed'].transform('mean'))


Empty DataFrame
Columns: [children, days_employed, dob_years, education, education_id, family_status, family_status_id, gender, income_type, debt, total_income, purpose, purp_category, age_group, income_group]
Index: []

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
purp_category       0
age_group           0
income_group        0
dtype: int64
Series([], Name: income_type, dtype: int64)
Series([], Name: children, dtype: int64)


### Вывод

Причина отсутствия данных не ясна. Больше похоже на случайные пропуски

<div class="alert alert-danger">

**Комментарий ревьюера:** 
* Ты корректно заменил значения в столбце с доходом, прошу тебя также обратить внимание на столбец `days_employed` и заменить пропуски в нем. Также можно разобраться с выбивающимися значениями в столбцах `days_employed` и `children`.
* Я заметил ты почти в каждой ячейке подгружаешь данные по новой. Во-первых, в этом нет необходимости, если ты подгрузил данные в начале они тебе потом доступны и дальше по проекту в рамках текущей сессии. Во-вторых, так ты перезаписываешь свою переменную `df` и тем самым отменяешь все операции, которые ты производил ранее. Это сказывается негативно на самом проекте и его целостности, поэтому попрошу тебя не подгружать тебя данные повторно. Тоже самое можно сказать о библиотеках, достаточно подгрузить их один раз в начале проекта.
</div>

<div class="alert alert-info">

* Очевидно, что причина в отсутвии данных в "days_employed" схожа с причиной отсуствия в столбце с доходом. В качестве практики заменил значения на средние по возрасту клиента. В принципе этот слобец не используется в дальшейшем, поэтому не рассматривал его. 
* По поводу "children" уже написал свое видение (шаг 1 - вывод п.5; шаг 3 - п.1)
* По поводу подгрузки данных по новой в каждой ячейке: тут невозможно с тобой не согласиться, поэтому поправил. Но, опираясь на свой небольшой опыт работы с JupiterNotebook, в некоторых случаях не совсем удобно разбивать код на несколько ячеек, т.к. при перезапуске ядра приходится воспроизводить отдельные ячейки, выполнение которых может потребовать время. Вижу здесь направление для своего развития.
    
</div>

<div class="alert alert-success">

**Комментарий ревьюера v2:** Да иногда перезагрузка ядра бывает не очень кстати, но зато можно комфортно работать в рамках одной сессии не перегружаясь. Со временем привыкнешь ко всем особенностям :)
</div>

<a id='3-bullet'></a>

### Замена типа данных

In [27]:
#Переведем столбец debt в тип boolean
df_debt_bool = df
df_debt_bool['debt'] = df_debt_bool['debt'].astype(bool)

print (df_debt_bool['debt'].value_counts())


False    19784
True      1741
Name: debt, dtype: int64


### Вывод

1741 человек имеют долг по кредиту. Около 8%

<a id='4-bullet'></a>

### Обработка дубликатов

In [28]:
#Приведем значения в столбце education к одному регистру и подсчитаем количество по каждому типу образования
df['education'] = df['education'].str.lower()
print(df.groupby('education').count().sort_values(by='children', ascending=False)['children'])


education
среднее                15233
высшее                  5260
неоконченное высшее      744
начальное                282
ученая степень             6
Name: children, dtype: int64


### Вывод

Больше всего берут кредиты люди со средним образованием и с высшим

<a id='5-bullet'></a>

### Лемматизация


In [29]:
#Приведем цели кредита к общим категориям: "ремонт", "недвижимость", "автомобиль", "образование", "свадьба"
#Для этого определим функцию создающую новый столбец

def purp_category(row):
    for word in row['purpose'].split(' '):
        stemmed_word = russian_stemmer.stem(word)
        if 'рем' in stemmed_word: return "ремонт"
        elif 'недв' in stemmed_word or 'жил' in stemmed_word: return "недвижимость"
        elif 'авто' in stemmed_word: return "автомобиль"
        elif 'образ' in stemmed_word: return "образование"
        elif 'свад' in stemmed_word: return "свадьба"

        
df['purp_category'] = df.apply(purp_category, axis=1)
print(df['purp_category'].value_counts())



недвижимость    10228
автомобиль       4315
образование      4022
свадьба          2348
ремонт            612
Name: purp_category, dtype: int64


### Вывод

Больше всего кредитов люди берут на приобритение недвижимости. 
На удивление 3е место с небольшим отрывом от 2го - это образование

<div class="alert alert-warning">

**Комментарий ревьюера:** Данная ячейка была оформлена как код и поэтому возникала ошибка. По правилам я должен был вернуть такой проект, но я решил переделать ее в markdown ячейку, чтобы не получилось так, что ты все это время ждал зря. Будь, пожалуйста, внимательнее следующий раз :)
</div>

<div class="alert alert-info"> 

***Благодарю!***

</div>

<a id='6-bullet'></a>

### Категоризация данных

In [30]:
#Попробуем сгруппировать данные столбца dob_years
def age_group(age):
    if age <= 18:
        return 'дети'
    elif age < 65:
        return 'взрослые'
    return 'пенсионеры'

df['age_group'] = df['dob_years'].apply(age_group)
print(df[df['days_employed'].isnull()]['age_group'].value_counts())
print(df['age_group'].value_counts())

#Разобьем людей на категории по уровню дохода предварительно переведем столбец total_income в integer type
#Т.к. выше не обнаружено убедительной причины пропуска данных в столбце с уровнем дохода, 
#для дальнейшего анализа придется исключить пропущенные данные
df = df.dropna(how='any')
print(df.info())

df['total_income'] = df['total_income'].astype(int)

def income_group(income):
    if income <= 50000:
        return 'невысокого достатка'
    elif income < 100000:
        return 'среднего достатка'
    elif income < 200000:
        return 'высокого достатка'
    return 'богатые'

df['income_group'] = df['total_income'].apply(income_group)
print(df['income_group'].value_counts())


Series([], Name: age_group, dtype: int64)
взрослые      20525
пенсионеры      899
дети            101
Name: age_group, dtype: int64
<class 'pandas.core.frame.DataFrame'>
Int64Index: 21525 entries, 0 to 21524
Data columns (total 14 columns):
children            21525 non-null int64
days_employed       21525 non-null float64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null bool
total_income        21525 non-null float64
purpose             21525 non-null object
purp_category       21525 non-null object
age_group           21525 non-null object
dtypes: bool(1), float64(2), int64(4), object(7)
memory usage: 2.3+ MB
None
высокого достатка      11995
богатые                 5067
среднего достатка       4091
невысокого д

### Вывод

И тут никакой корреляции с отсутвующими данными в столбцах "days_employed" и "total_income"
Самая большая группа кредитуемых - это люди от 18 до 65 лет. 

<div class="alert alert-warning">

**Комментарий ревьюера:** Хорошо бы произвести категоризацию по доходу также в данном разделе для целостности проекта и структуры.
</div>

### Шаг 3. Ответьте на вопросы

<a id='7-bullet'></a>

- Есть ли зависимость между наличием детей и возвратом кредита в срок?

In [31]:
#Проведем предобработку по столбцу "children". Как описано выше 76 человек имеют 20 детей и 47 человек по -1 ребенку
#Т.к. данных немного позволим себе убрать эти строки из таблицы для дальнейшего анализа

df = df[(df['children']!=-1) & (df['children']!=20)]
print(df['children'].value_counts())

#Агрегируем данные группировки по количеству детей со столбцом "debt"
children_debt = df.groupby('children').agg({'debt': ['sum', 'count']})
print(children_debt)

#Найдем процент должников в каждой группе по количеству детей
children_debt['debt_perc'] = children_debt['debt']['sum'] / children_debt['debt']['count'] * 100
print(children_debt.sort_values(by='debt_perc', ascending=False))


0    14149
1     4818
2     2055
3      330
4       41
5        9
Name: children, dtype: int64
            debt       
             sum  count
children               
0         1063.0  14149
1          444.0   4818
2          194.0   2055
3           27.0    330
4            4.0     41
5            0.0      9
            debt        debt_perc
             sum  count          
children                         
4            4.0     41  9.756098
2          194.0   2055  9.440389
1          444.0   4818  9.215442
3           27.0    330  8.181818
0         1063.0  14149  7.512898
5            0.0      9  0.000000


### Вывод

Бездетные явно в лидерах по возврату кредита. На 100 бездетных приходится 7.5 невозвратов. Стопроцентный возрат по категории "5 детей" скорее всего связан с небольшим количеством таких людей в выборке.

<a id='8-bullet'></a>

- Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [32]:
#Произведем группировку и аггрегацию по аналогии с предыдущим заданием
family_debt = df.groupby('family_status').agg({'debt': ['sum', 'count']})
print(family_debt)

#Найдем процент должников в каждой группе по семейному положению
family_debt['debt_perc'] = family_debt['debt']['sum'] / family_debt['debt']['count'] * 100
print(family_debt.sort_values(by='debt_perc', ascending=False))

                        debt       
                         sum  count
family_status                      
Не женат / не замужем  273.0   2799
в разводе               84.0   1189
вдовец / вдова          63.0    952
гражданский брак       385.0   4160
женат / замужем        927.0  12302
                        debt        debt_perc
                         sum  count          
family_status                                
Не женат / не замужем  273.0   2799  9.753483
гражданский брак       385.0   4160  9.254808
женат / замужем        927.0  12302  7.535360
в разводе               84.0   1189  7.064760
вдовец / вдова          63.0    952  6.617647


### Вывод

В антилидерах те, кто никогда не заключал официального брака: больше 9 случаев невозврата из 100. 
Надежнее всего кредитовать овдовевших и разведенных людей.

<a id='9-bullet'></a>

- Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [33]:
#Произведем группировку и аггрегацию по аналогии с предыдущим заданием
income_debt = df.groupby('income_group').agg({'debt': ['sum', 'count']})
print(income_debt)

#Найдем процент должников в каждой группе по уровню дохода
income_debt['debt_perc'] = income_debt['debt']['sum'] / income_debt['debt']['count'] * 100
display(income_debt.sort_values(by='debt_perc', ascending=False))

                       debt       
                        sum  count
income_group                      
богатые               356.0   5039
высокого достатка    1022.0  11919
невысокого достатка    23.0    371
среднего достатка     331.0   4073


debt        debt_perc
                        sum  count          
income_group                                
высокого достатка    1022.0  11919  8.574545
среднего достатка     331.0   4073  8.126688
богатые               356.0   5039  7.064894
невысокого достатка    23.0    371  6.199461

### Вывод

Люди высокого достатка (100-200 т.р. в мес.) самые злостные неплательщики. 
Самые ответственные оказались люди невысокого достатка

<a id='10-bullet'></a>

- Как разные цели кредита влияют на его возврат в срок?

In [34]:
#У нас уже есть обобщенные категории целей кредитования. Осталось проделать действия из предыдущих заданий
#Произведем группировку и аггрегацию
purpose_debt = df.groupby('purp_category').agg({'debt': ['sum', 'count']})
print(purpose_debt)

#Найдем процент должников в каждой группе по цели кредитования
purpose_debt['debt_perc'] = purpose_debt['debt']['sum'] / purpose_debt['debt']['count'] * 100
display(purpose_debt.sort_values(by='debt_perc', ascending=False))


                debt       
                 sum  count
purp_category              
автомобиль     400.0   4288
недвижимость   745.0  10171
образование    369.0   3997
ремонт          35.0    609
свадьба        183.0   2337


debt        debt_perc
                 sum  count          
purp_category                        
автомобиль     400.0   4288  9.328358
образование    369.0   3997  9.231924
свадьба        183.0   2337  7.830552
недвижимость   745.0  10171  7.324747
ремонт          35.0    609  5.747126

### Вывод

Автомобили и образование заставляют людей чаще не платить во время по кредитам

<div class="alert alert-warning">

**Комментарий ревьюера:** В целом по данному разделу все ок, но можно немного его подправить, если учесть комментарии, которые я делал выше. Например, таблицы можно выводить с помощью `display()`.
</div>

<a id='11-bullet'></a>

### Шаг 4. Общий вывод

Бездетные люди чаще в срок возвращают кредит. При этом люди в официальном браке (или хотя бы побывавшие в нем) также более желаемые клиенты кредитной организации.
Таким образом можно предположить, что наиболее надежными заемщиками можно считать бездетных и овдовевших людей

<div class="alert alert-info">
    
1. Отсутствующие данные в столбцах "days_employed" и "total_income" можно объяснить случайным отсутствием. В качестве практики заменил стаж на среднее в зависимости от возраста, а доход на медиану в зависимости от типа занятости.
2. Отрицательные значения в столбце "days_employed" можно объяснить заменой знака дефиса на знак минус.
3. В столбце с количеством детей встречаются аномальные значения. Но ввиду того, их количество ничтожно мало по сравнению с количеством строк всей таблицы, то их можно просто исключить
4. Кредиты берут в основном люди со средним и высшим образованием. Основыние цели - недвижимость, авто и на удивление образование
5. Наиболее надежными заемщиками стали группы: без детей, состоящие или состоявшие в официалном браке, с невысоким доходом (менее 50т.р.). Наименее рискованная цель кредита - ремонт
    
</div>

<div class="alert alert-danger">

**Комментарий ревьюера:** Предлагаю немного расширить вывод и добавить в него краткий перечень выполненных шагов и находок.
</div>

<div class="alert alert-info">

Алан, благодарю за твою помошь!

</div>

<div class="alert alert-danger">

**Вывод:** Ты проделал большую работу и у тебя получился достойный проект, но мы можем его улучшить для этого нам осталось:
* Оформить вводную часть проекта
* Обработать все пропуски и выбивающиеся значения
* Не подгружать данные повторно 
* Расширить вывод
</div>

<div class="alert alert-success">

**Вывод v2:** Спасибо, что учел все мои комментарии, было приятно с тобой поработать и я рад, что нам удалось улучшить твой проект :) Поздравляю с завершением первого проекта и желаю тебе дальнейших успехов! :)
</div>

### Чек-лист готовности проекта

Поставьте 'x' в выполненных пунктах. Далее нажмите Shift+Enter.

- [x]  открыт файл;
- [x]  файл изучен;
- [x]  определены пропущенные значения;
- [x]  заполнены пропущенные значения;
- [x]  есть пояснение, какие пропущенные значения обнаружены;
- [x]  описаны возможные причины появления пропусков в данных;
- [x]  объяснено, по какому принципу заполнены пропуски;
- [x]  заменен вещественный тип данных на целочисленный;
- [x]  есть пояснение, какой метод используется для изменения типа данных и почему;
- [x]  удалены дубликаты;
- [x]  есть пояснение, какой метод используется для поиска и удаления дубликатов;
- [x]  описаны возможные причины появления дубликатов в данных;
- [x]  выделены леммы в значениях столбца с целями получения кредита;
- [x]  описан процесс лемматизации;
- [x]  данные категоризированы;
- [x]  есть объяснение принципа категоризации данных;
- [x]  есть ответ на вопрос: "Есть ли зависимость между наличием детей и возвратом кредита в срок?";
- [x]  есть ответ на вопрос: "Есть ли зависимость между семейным положением и возвратом кредита в срок?";
- [x]  есть ответ на вопрос: "Есть ли зависимость между уровнем дохода и возвратом кредита в срок?";
- [x]  есть ответ на вопрос: "Как разные цели кредита влияют на его возврат в срок?";
- [x]  в каждом этапе есть выводы;
- [x]  есть общий вывод.